src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,147 @@
+        }
+        File[] saves = dir.listFiles();
+        String[] filenames = new String[saves.length];
+        for (int i = 0; i < saves.length; i++) {
+            filenames[i] = saves[i].getName();
+        }
+        return filenames;
+    }
+
+    /**
+     * Saves a <code>TaskList</code> to a file, in a loadable format.
+     *
+     * @param pathString path of the file to save to.
+     * @param tasks      tasklist to save.
+     * @throws IOException if the file could not be written to.
+     */
+    public void save(String pathString, TaskList tasks) throws IOException {
+        createSaveDir();
+        Path path = FileSystems.getDefault().getPath(dataPath).resolve(pathString);
+        BufferedWriter bw = Files.newBufferedWriter(path, StandardCharsets.UTF_8);
+        try {
+            bw.write(Integer.toString(tasks.size()));
+            for (int i = 0; i < tasks.size(); i++) {
+                String[] repres = tasks.get(i).save();
+                bw.newLine();
+                bw.write(Integer.toString(repres.length));
+                for (int j = 0; j < repres.length; j++) {
+                    bw.newLine();
+                    bw.write(repres[j]);
+                }
+            }
+        } finally {
+            bw.close();
+        }
+    }
+
+    /**
+     * Loads a file as a <code>TaskList</code>.
+     *
+     * @param pathString path of the file to load from.
+     * @param tasks      tasklist to load to.
+     * @throws IOException       if the file could not be read from.
+     * @throws MikiLoadException if the file did not represent a correctly-encoded TaskList.
+     */
+    public void load(String pathString, TaskList tasks) throws IOException, MikiLoadException {
+        createSaveDir();
+        tasks.clear();
+        Path path = FileSystems.getDefault().getPath(dataPath).resolve(pathString);
+        BufferedReader br = Files.newBufferedReader(path, StandardCharsets.UTF_8);
+        try {
+            int numTasks = Integer.parseInt(br.readLine());
+            for (int i = 0; i < numTasks; i++) {
+                int taskLines = Integer.parseInt(br.readLine());
+                String[] repres = new String[taskLines];
+                for (int j = 0; j < taskLines; j++) {
+                    repres[j] = br.readLine();
+                }
+                switch (repres[0].charAt(0)) {
+                case 'T':
+                    tasks.add(Todo.parseLoad(repres));
+                    break;
+                case 'D':
+                    tasks.add(Deadline.parseLoad(repres));
+                    break;
+                case 'E':
+                    tasks.add(Event.parseLoad(repres));
+                    break;
+                default:
+                    //back-compat? try to handle
+                }
+            }
+        } catch (NumberFormatException | TaskParseException ex) {
+            throw new MikiLoadException("this file is corrupt...");
```

Comment:
Corrupted instead of corrupt?

Reply:
👀 Both are grammatical actually, but looks like corrupted is a lot more common when referring to files so switched to that